### PR TITLE
refactor: remove unnecessary hook

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR removes a Hook that is pretty much useless given we have the `ActiveLink` component; This PR also updates the ActiveLink component to have same logic features as the previous Hook and updates the SideNavigation Component making it RSC-compatible and subsequently making the DocsLayout compatible.

This PR also updates related styles as now the "active" class is on the link.

## Validation

Deployment should work; Docs pages should work; Active Links should have active styling as before (Header and Side Navigation)

## Related Issues

https://github.com/amannn/next-intl/issues/611

### CODE REVIEW

**Comment by ovflowd** [comment]
cc @nodejs/nodejs-website requesting this as a fast-track mostly because it's supposed to be a small refactor/fix to unblock other components

File: util/getBitness.ts [hunk 1]
```diff
@@ -9,10 +9,8 @@ export const getBitness = async () => {
       'bitness',
     ]);
 
-    // Apparently in some cases this is not a Promise, then we should ignore
-    if (entropyValues instanceof Promise) {
-      return entropyValues.then(ua => ua.bitness);
-    }
+    // Apparently in some cases this is not a Promise, we can Promisify it.
+    return Promise.resolve(entropyValues).then(ua => ua.bitness);
```
Comment (canerakdas) [inline util/getBitness.ts]: When the promise returns `undefined`, `ua.bitness` can throw an error

```suggestion
    return Promise.resolve(entropyValues)
      .then(({ bitness }) => bitness)
      .catch(() => undefined);
```

or we can wrap it with try catch

Comment (ovflowd) [reply util/getBitness.ts]: > When the promise returns undefined, ua.bitness can throw an error

Right, that's what I was saying to @ljharb 👀

**Review by AugustinMauroy — APPROVED** [review]
SGTM !

**Review by canerakdas — APPROVED** [review]
LGTM!

**Pull request merged**